python_serviceplatformen/authentication.py

```diff
@@ -44,27 +46,49 @@ def get_access_token(self, entity_id: str) -> str:
 
         Args:
             entity_id: The entity id of the endpoint.
-            test: Whether to use the test api or not.
 
         Returns:
             An access token to be used in api calls.
 
         Raises:
-            HTTPError: If an access token couldn't be obtained for the given entity id.
+            ValueError: If an access token couldn't be obtained for the given entity id.
         """
         if entity_id in self._access_tokens and datetime.now() < self._access_tokens[entity_id][1]:
             return self._access_tokens[entity_id][0]
 
         try:
-            saml_token = _get_saml_token(self.cvr, self.cert_path, entity_id, test=self.test)
+            saml_token = self.get_saml_token(entity_id)
             access_token = _get_access_token(saml_token, self.cert_path, test=self.test)
             self._access_tokens[entity_id] = access_token
             return access_token[0]
         except HTTPError as exc:
             raise ValueError(f"Couldn't obtain access token for {entity_id}: {exc.response.text}") from exc
 
+    def get_saml_token(self, entity_id: str) -> str:
+        """Get a SAML to the api endpoint with the given entity id.
+        If a SAML token already exists for the endpoint it is reused.
+
+        Args:
+            entity_id: The entity id of the endpoint.
+
+        Raises:
+            ValueError: If a SAML token couldn't be obtained for the given entity id.
+
+        Returns:
+            The SAML token as a string
+        """
+        if entity_id in self._saml_tokens and datetime.now() < self._saml_tokens[entity_id][1]:
+            return self._saml_tokens[entity_id][0]
```

Comment:
For improved readability, you could use constants for `0` and `1` here, e.g.

```suggestion
        if entity_id in self._saml_tokens and datetime.now() < self._saml_tokens[entity_id][EXPIRES_AT]:
            return self._saml_tokens[entity_id][TOKEN]
```